src/aiida/transports/plugins/ssh_async.py

```diff
@@ -0,0 +1,1272 @@
+                'The script should be executable',
+                'non_interactive_default': True,
+                'callback': validate_script,
+            },
+        ),
+    ]
+
+    @classmethod
+    def _get_machine_suggestion_string(cls, computer):
+        """Return a suggestion for the parameter machine."""
+        # Originally set as 'Hostname' during `verdi computer setup`
+        # and is passed as `machine=computer.hostname` in the codebase
+        # unfortunately, name of hostname and machine are used interchangeably in the aiida-core codebase
+        # TODO: open an issue to unify the naming
+        return computer.hostname
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # the machine is passed as `machine=computer.hostname` in the codebase
+        # 'machine' is immutable.
+        # 'machine_or_host' is mutable, so it can be changed via command:
+        # 'verdi computer configure core.ssh_async <LABEL>'.
+        # by default, 'machine_or_host' is set to 'machine' in the __init__ method, if not provided.
+        # NOTE: to guarantee a connection,
+        # a computer with core.ssh_async transport plugin should be configured before any instantiation.
+        self.machine = kwargs.pop('machine_or_host', kwargs.pop('machine'))
+        self._max_io_allowed = kwargs.pop('max_io_allowed', self._DEFAULT_max_io_allowed)
+        self.script_before = kwargs.pop('script_before', 'None')
+
+        self._concurrent_io = 0
+
+    @property
+    def max_io_allowed(self):
+        return self._max_io_allowed
+
+    async def _lock(self, sleep_time=0.5):
+        while self._concurrent_io >= self.max_io_allowed:
+            await asyncio.sleep(sleep_time)
+        self._concurrent_io += 1
+
+    async def _unlock(self):
+        self._concurrent_io -= 1
+
+    async def open_async(self):
+        """Open the transport.
+        This plugin supports running scripts before and during the connection.
+        The scripts are run locally, not on the remote machine.
+
+        :raises InvalidOperation: if the transport is already open
+        """
+        if self._is_open:
+            raise InvalidOperation('Cannot open the transport twice')
+
+        if self.script_before != 'None':
+            os.system(f'{self.script_before}')
+
+        self._conn = await asyncssh.connect(self.machine)
+        self._sftp = await self._conn.start_sftp_client()
+
+        self._is_open = True
+
+        return self
+
+    async def close_async(self):
+        """Close the transport.
+
+        :raises InvalidOperation: if the transport is already closed
+        """
+        if not self._is_open:
+            raise InvalidOperation('Cannot close the transport: it is already closed')
+
+        self._conn.close()
+        await self._conn.wait_closed()
+        self._is_open = False
+
+    def chown(self, path, uid, gid):
+        raise NotImplementedError
```

Comment:
Don't we support `chown` in `ssh_async`? There is a `chown_async` method and by inheriting from `AsyncTransport` we automatically use it with `run_command_blocking`?